modules/core/04-channel/v2/keeper/msg_server.go

```diff
@@ -173,10 +185,24 @@ func (k *Keeper) Acknowledgement(ctx context.Context, msg *types.MsgAcknowledgem
 		return nil, errorsmod.Wrap(err, "acknowledge packet verification failed")
 	}
 
+	// acknowledgement only needs to be filled in if the receive was successful
+	// thus we only validate if receive is successful
+	if msg.Acknowledgement.RecvSuccess {
```

Comment:
One annoying thing here is I prevent an empty ack in RecvPacket or WriteAck since this could be a developer error

But on AcknowledgePacket, I allow relayers to just pass in empty ack to signal a failed receive since the alternative is to enforce a sentinel non-empty ack that will not be inspected at all by handler

Reply:
Open to feedback on better ways to structure